cli/src/commands/config.ts

```diff
@@ -0,0 +1,245 @@
+            process.exit(1);
+          }
+          setEnvironment(options.env as Environment);
+          printSuccess(`Environment set to: ${options.env}`);
+        }
+
+        // If --show or no options, display current config
+        if (
+          options.show ||
+          (!options.apiKey && !options.env && !options.clear)
+        ) {
+          printCurrentConfig();
+        }
+      } catch (error: unknown) {
+        const message =
+          error instanceof Error ? error.message : "Unknown error";
+        printError(`Config error: ${message}`);
+        process.exit(1);
+      }
+    });
+
+  // Add login subcommand
+  cmd
+    .command("login")
+    .description("Interactive login setup")
+    .action(async () => {
+      try {
+        console.log(`\n${symbols.key} ${chalk.bold("DeSci Nodes CLI Login")}\n`);
+
+        // Select environment
+        const currentEnv = getEnvironment();
+        const envChoices = [
+          {
+            name: "dev",
+            message: `Development ${chalk.dim("(nodes-dev.desci.com)")}`,
+            value: "dev",
+          },
+          {
+            name: "staging",
+            message: `Staging ${chalk.dim("(nodes-staging.desci.com)")}`,
+            value: "staging",
+          },
+          {
+            name: "prod",
+            message: `Production ${chalk.dim("(nodes.desci.com)")}`,
+            value: "prod",
+          },
+          {
+            name: "local",
+            message: `Local ${chalk.dim("(localhost:5420)")}`,
+            value: "local",
+          },
+        ];
+
+        const env = await select<Environment>({
+          message: "Select environment:",
+          choices: envChoices,
+          initial:
+            currentEnv === "dev"
+              ? 0
+              : currentEnv === "staging"
+                ? 1
+                : currentEnv === "prod"
+                  ? 2
+                  : 3,
+        });
+
+        setEnvironment(env);
+
+        const webUrl = WEB_URLS[env];
+        console.log(
+          `\n${symbols.info} Get your API key from: ${chalk.cyan(webUrl)}`,
+        );
+        console.log(chalk.dim("  Go to Profile → API Keys → Create New Key\n"));
+
+        // Get API key (masked with *)
+        const apiKey = await password({
+          message: "Enter your API key:",
+          validate: (value: string) => {
+            if (!value || value.length < 10) {
+              return "Please enter a valid API key";
+            }
+            return true;
+          },
+        });
+
+        setApiKey(apiKey);
+
+        // Optionally set private key for publishing
+        console.log(
+          chalk.dim("\nOptional: Set up a private key for publishing nodes."),
+        );
+        console.log(
+          chalk.dim("This allows you to sign and publish directly from the CLI.\n"),
+        );
+
+        const setupPrivateKey = await confirm({
+          message: "Do you want to set up a private key for publishing?",
```

Comment:
Perhaps make it clear that it's never sent to the server, only used to sign locally. If it is unset in config, perhaps the publish function could prompt for it when needed, so it's only kept in memory and not left laying around on disk unencrypted.